pareto/strategic_water_management/strategic_produced_water_optimization.py

```diff
@@ -3402,6 +3488,454 @@ def SeismicActivityExceptionRule(model, k, t):
+        )
+        mh.p_rhog = Param(
+            initialize=9.8 * 1000,
+            units=model.model_units["pressure"] / pyunits.meter,
+            doc="g (m/s2) * density of PW (assumed to be 1000 kg/m3)",
+        )
+        mh.p_Min_AOP = Param(
+            initialize=10 * 6895,
+            mutable=True,
+            units=model.model_units["pressure"],
+            doc="Minimum ALlowable Operating Pressure 14.7 psi, in Pa",
+        )
+        mh.p_Max_AOP = Param(
+            initialize=150 * 6895,
+            mutable=True,
+            units=model.model_units["pressure"],
+            doc="Maximum ALlowable Operating Pressure 150 psi, in Pa",
+        )
+        mh.p_WellPressure = Param(
+            model.s_P,
+            model.s_T,
+            default=0,
+            initialize={
+                key: pyunits.convert_value(
+                    value,
+                    from_units=pyunits.psi,
+                    to_units=model.model_units["pressure"],
+                )
+                for key, value in model.df_parameters["WellPressure"].items()
+            },
+            units=model.model_units["pressure"],
+            doc="Well pressure at production or completions pad [pressure]",
+        )
+
+        mh.v_Pressure = Var(
+            model.s_L,
+            model.s_T,
+            within=NonNegativeReals,
+            initialize=mh.p_Min_AOP,
+            bounds=(mh.p_Min_AOP, None),
+            units=model.model_units["pressure"],
+            doc="Pressure at location l at time t in Pa",
+        )
+        for p in model.s_P:
+            for t in model.s_T:
+                if value(mh.p_WellPressure[p, t]) > 0:
+                    mh.v_Pressure[p, t].fix(mh.p_WellPressure[p, t])
+
+        mh.p_diameter_Pipeline_eff = Param(
+            model.s_LLA,
+            default=0,
+            initialize=0,
+            mutable=True,
+            units=model.model_units["diameter"],
+            doc="Diameter of pipeline between two locations [inch]",
+        )
+        mh.p_HW_loss = Param(
+            model.s_LLA,
+            model.s_T,
+            default=0,
+            initialize=0,
+            within=NonNegativeReals,
+            mutable=True,
+            units=pyunits.meter,
+            doc="Hazen-Williams Frictional loss, m",
+        )
+        for key in model.s_LLA:
+            mh.p_diameter_Pipeline_eff[key] = model.df_parameters[
+                "InitialPipelineDiameters"
+            ][key] + sum(
+                model.vb_y_Pipeline[key, d]
+                * model.df_parameters["PipelineDiameterValues"][d]
+                for d in model.s_D
+            )
+
+        # Compute Hazen Williams head
+        for t0 in model.s_T:
+            for key in model.s_LLA:
+                if value(model.v_F_Piped[key, t0]) > 0.01:
+                    if value(mh.p_diameter_Pipeline_eff[key]) > 0.1:
+                        mh.p_HW_loss[key, t0] = _hazen_williams_head(
```

Comment:
The first three arguments passed to `_hazen_williams_head` are Pyomo expressions, but the fourth argument is a float. So, I think then `_hazen_williams_head` will return a Pyomo expression. Is that what you want? Or does it need to return a float instead? I am not sure here.

Reply:
The method is now updated for consistency. However, in either case, the method `_hazen_williams_head` only returns a float

Reply:
Resolving.